### PR TITLE
Bring up services reaaal slooow

### PR DESCRIPTION
**Pull request by stkem**
@andrewconner @eishay have a quick look please.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala [hunk 1]
```diff
@@ -58,8 +60,13 @@ abstract class FortyTwoGlobal(val mode: Mode.Mode)
       Statsd.increment("deploys", 42)
       injector.instance[AirbrakeNotifier].reportDeployment()
       injector.instance[HealthcheckPlugin].reportStart()
-      injector.instance[HealthcheckPlugin].warmUp()
+      injector.instance[HealthcheckPlugin].warmUp(injector.instance[BenchmarkRunner])
     }
+    val selfCheckPassed: Boolean = Await.result(serviceDiscovery.startSelfCheck(), Duration.Inf)
+    if (!selfCheckPassed) {
+      log.error("STARTUP SELF CHECK FAILED!")
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala:67]: We'd never see this. Needs Airbrake / email / smoke signal

File: kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala [hunk 2]
```diff
@@ -224,5 +224,14 @@ class HealthcheckPluginImpl @Inject() (
     email
   }
 
-  override def warmUp() = scheduleTaskOnce(actor.system, 3 minutes, "Healthcheck: consider service warm") {super.warmUp()}
+  override def warmUp(benchmarkRunner: BenchmarkRunner) : Unit = {
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:227]: This seems to run in dev too?

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:227]: onStart only calls this in prod

File: kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala:67]: Well, the server will fail to go into "UP" mode, causing the deploy script to bring it back down.

**Pull request merged**